elastic/kibana
# [uiSettings] Fix validation error for 0 and false value

## Summary

This PR fixes the condition for whether we should throw a `bad value` error from the `validate` uiSettings api.
Currently, the error would be thrown for values `0` and `false`, which is incorrect behaviour as these are valid values. We should change the condition to only catch when the value is `null` or `undefined`.


**How to test:**
We can test this by inputting `0` or `false` in a setting input where these values are invalid. For example, the APM setting `apmServiceGroupMaxNumberOfServices` has `schema` restriction for minimum value `1`: https://github.com/elastic/kibana/blob/b4d93fc145c3c09eb1096c610b7cd736f19f6a3a/x-pack/plugins/observability/server/ui_settings.ts#L189

1. Start Es with `yarn es snapshot` and Kibana with `yarn start`.
2. Go to APM -> General setting
3. Change the `apmServiceGroupMaxNumberOfServices` setting to `0`.
4. Verify that a validation error message shows up.
